pkg/runctl/run_manager.go

```diff
@@ -366,7 +366,17 @@ func (c *runManager) setupNetworkPolicyFromConfig(ctx *runContext) error {
 		Kind:  "NetworkPolicy",
 	}
 
-	configStr := c.pipelineRunsConfig.NetworkPolicy
+	configStr := c.pipelineRunsConfig.DefaultNetworkPolicy
+
+	spec := ctx.pipelineRun.GetSpec()
+	if spec.Profiles != nil && spec.Profiles.Network != "" {
+		configStr = c.pipelineRunsConfig.NetworkPolicies[spec.Profiles.Network]
+		if configStr == "" {
+			ctx.pipelineRun.UpdateResult(v1alpha1.ResultErrorConfig)
+			return fmt.Errorf("Cannot find network policy with name %q in network config map", spec.Profiles.Network)
+		}
+	}
```

Comment:
If the first `if` is not entered, line 384 tries to create a network policy using the _default network policy key_ as manifest! This is just plain wrong. 

The code must be overhauled. Especially, do not re-use variables for other purposes.

Reply:
c.pipelineRunsConfig.DefaultNetworkPolicy is already the policy and not the key.
The policy is already loaded in the config.go

Reply:
ok